src/engine/util/util_load_save.cpp

```diff
@@ -32,6 +32,24 @@ Tilemap_save_to_file(const char *filename, Tilemap tilemap)
     return true;
 }
 
+// -----------------------------------------------------------------
+bool 
+TilesetSaveToFile(const char* filename, Tileset tileset)
+{
+    SDL_RWops *write_context = SDL_RWFromFile(filename, "wb");
+    if (write_context == nullptr)
+    {
+        return false;
+    }
+
+    nlohmann::json tileset_json;
+    nlohmann::to_json(tileset_json, tileset);
+    std::string tileset_json_str = tileset_json.dump();
+    write_context->write(write_context, tileset_json_str.c_str(), sizeof(char), tileset_json_str.size());
+
```

Comment:
context should be closed here with `write_context->close()` or `SDL_RWclose()`.
I forgot about this in the Tilemap save function so please add this there too. thanks :)